core/src/main/java/com/google/instrumentation/stats/Distribution.java

```diff
@@ -0,0 +1,236 @@
+  /**
+   * Constructs a new {@link Distribution} with specified {@code bucketBoundaries}.
+   *
+   * @param bucketBoundaries the boundaries for the buckets in the underlying {@code Distribution}.
+   * @return a new, empty {@code Distribution} with the specified boundaries.
+   * @throws NullPointerException if {@code bucketBoundaries} is null.
+   * @throws IllegalArgumentException if {@code bucketBoundaries} is not sorted, or has zero length.
+   */
+  public static final Distribution create(List<Double> bucketBoundaries) {
+    checkNotNull(bucketBoundaries, "bucketBoundaries");
+    checkArgument(bucketBoundaries.size() > 0, "Zero length bucket boundaries");
+    double lower = bucketBoundaries.get(0);
+    for (int i = 1; i < bucketBoundaries.size(); i++) {
+      double next = bucketBoundaries.get(i);
+      checkArgument(lower < next, "Bucket boundaries not sorted.");
+      lower = next;
+    }
+    return new Distribution(bucketBoundaries);
+  }
+
+  // Returns true if the distribution has histogram buckets.
+  private boolean hasBuckets() {
+    return (bucketBoundaries != null);
+  }
+
+  // Construct a new Distribution with given boundaries.
+  private Distribution(List<Double> bucketBoundaries) {
+    this.bucketBoundaries =
+        (bucketBoundaries == null) ? null : new ArrayList<Double>(bucketBoundaries.size());
+    if (hasBuckets()) {
+      for (Double value : bucketBoundaries) {
+        this.bucketBoundaries.add(value);
+      }
+      bucketCounts = new Long[bucketBoundaries.size() + 1];
+      for (int i = 0; i < bucketCounts.length; i++) {
+        bucketCounts[i] = 0L;
+      }
+    }
+  }
+
+  /**
+   * The number of values in the population. Must be non-negative.
+   *
+   * @return The number of values in the population.
+   */
+  public long getCount() {
+    return count;
+  }
+
+  /**
+   * The arithmetic mean of the values in the population. If {@link #getCount()} is zero then this
+   * value will be NaN.
+   *
+   * @return The arithmetic mean of all values in the population.
+   */
+  public double getMean() {
+    return sum / count;
+  }
+
+  /**
+   * The sum of the values in the population. If {@link #getCount()} is zero then this value must
+   * also be zero.
+   *
+   * @return The sum of values in the population.
+   */
+  public double getSum() {
+    return sum;
+  }
+
+  /**
+   * The range of the population values. If {@link #getCount()} is zero then this returned range is
+   * implementation-dependent. @ return The {code Range} representing the range of population
+   * boundaries.
+   */
+  public Range getRange() {
+    return range;
+  }
+
+  /**
+   * A Distribution may optionally contain a histogram of the values in the population. The
+   * histogram is given in {@link #getBucketCounts()} as counts of values that fall into one of a
+   * sequence of non-overlapping buckets, described by {@link
+   * DistributionAggregationDescriptor#getBucketBoundaries()}. The sum of the values in {@link
```

Comment:
Should the link point to Distribution instead of DistributionAggregationDescriptor, since the Javadoc for Distribution also explains the buckets?